### PR TITLE
Settings and initial contents for Chinese localization

### PR DESCRIPTION
**Pull request by hanyuancheung**
This PR is to initiate Chinese localization based on
- [x] Issue to #451 
- [x] https://github.com/cncf/glossary/blob/main/LOCALIZATION.md guide

Base and target branch of this PR is dev-zh which is the development branch for Chinese localization.

This PR includes

- [languages.zh] section in config.toml
- content/zh/ directory
- i18n/zh.toml
- Some translated documents
  - Main page
    - content/zh/_index.md
  - Terms
    - "Cloud Native Technology": content/zh/cloud_native_tech.md
  - Others
    - contribute: content/zh/contribute/_index.md (partially localized)
    - style-guide: content/zh/style-guide/_index.md (partially localized)

Since the Chinese localization team owns content/zh/* only,
this PR will require approvals from upstream owners (maintainers).

### CODE REVIEW

**Comment by seokho-son** [comment]
@hanyuancheung  ! Thank you for the contribution ! :)

We are going to check following tasks in this PR. (target branch is dev-zh)

- [ ] 3-1. Adding a new language setting to site configuration https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-1-adding-a-new-language-setting-to-site-configuration
- [ ] 3-2. Adding a file for site strings https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-2-adding-a-file-for-site-strings
- [ ] 3-3. Adding a new localization directory https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-3-adding-a-new-localization-directory
- [ ] 3-4. Minimum required content https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-4-minimum-required-content
- [ ] 3-5. Check configuration works https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-5-check-configuration-works-and-open-a-pr

In the meantime, 
1) Since I cannot review Chinese contents, could you invite volunteers who can review Chinese contents in this PR ? (I expect two reviewers' LGTM)
2) Cloud you test Hugo Build locally for this PR to verify the configuration works ? (This page does not have Netlify preview yet)

File: content/zh/style-guide/_index.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+title: 风格指南
+toc_hide: true
+menu:
+main:
+weight: 10
+pre: <i class='fas fa-ruler-horizontal'></i>
+---
```
Comment (seokho-son) [inline content/zh/style-guide/_index.md:8]: I think this part will bring build failure :)

Comment (hanyuancheung) [reply content/zh/style-guide/_index.md]: yes, I've fixed its format.

**Comment by hanyuancheung** [comment]
截屏2022-02-22 上午11 19 59

I've tested Hugo Build locally and successfully.
Cause I don't have the permission to invite reviewers, @seokho-son could you please help to add the approver permission? I'll invite and inform them to review this PR. Thx.

**Comment by seokho-son** [comment]
@hanyuancheung 

> I've tested Hugo Build locally and successfully.

Thanks !! :) We enabled Netlify preview as well.
https://deploy-preview-464--cncfglossary.netlify.app/zh/


> Cause I don't have the permission to invite reviewers, @seokho-son could you please help to add the approver permission? I'll invite and inform them to review this PR. Thx.

We have additional procedure to give a permission :)
(need a consensus from maintainers first. I am going to arrange it)

The reviews I requested is about having comments, discussion, and consensus among Chinese volunteers using Comments in this PR. The volunteers can leave `LGTM` comment. don't need to approve this PR. 
With LGTM comments, I will assume the Chinese contents in this PR has some agreement :)

(We hope to see some activities of volunteers who will apply for approvers)

**Review by Jacob953 — APPROVED** [review]
LGTM.

**Comment by Rocksnake** [comment]
approve

**Review by Rocksnake — COMMENTED** [review]
LGTM.

**Comment by Submarinee** [comment]
LGTM.

**Comment by hanyuancheung** [comment]
@seokho-son I found that the approvers are added successfully. Could this PR also be merged?

**Review by Jacob953 — APPROVED** [review]
LGTM.

**Review by Rocksnake — APPROVED** [review]
LGTM.

**Pull request merged**